### PR TITLE
WIP: Add support for http protocol requests

### PR DESCRIPTION
**Pull request by nechaido**
- [ ] tests are added/updated for bug fixes and new features
- [x] description of changes is added in CHANGELOG.md

Currently, request to `http://localhost/api` would be transformed into requests to `https://localhost/api`, this will result in an obvious but hard-to-track OpenSSL error.

### CODE REVIEW

**Review by tshemsedinov — APPROVED** [review]

**Comment by tshemsedinov** [comment]
Debased and simplified, ready for landing @nechaido

File: lib/client.js [hunk 1]
```diff
@@ -92,7 +92,7 @@ class Metacom extends EventEmitter {
         const target = interfaceName + '/' + methodName;
         const packet = { call: callId, [target]: args };
         const dest = new URL(this.url);
-        const protocol = dest.protocol === 'ws:' ? 'http' : 'https';
+        const protocol = dest.protocol.endsWith('s') ? 'https' : 'http';
```
Comment (nechaido) [inline lib/client.js:95]: @tshemsedinov both `ws` and `wss` end with `s` that's why I used the check I used.

**Comment by tshemsedinov** [comment]
See client-side implementation: https://github.com/metarhia/metacom/tree/master/dist

**Comment by nechaido** [comment]
I'd prefer to have the same version for both node and browser clients.

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,7 @@
 ## [Unreleased][unreleased]
 
 - Fix conversion of custom thrown exceptions for response to client
+- Add support for `http ` protocol requests
```
Comment (tshemsedinov) [inline CHANGELOG.md:6]:
```suggestion
- Add support for `http` protocol requests
```

**Comment by tshemsedinov** [comment]
Looks like not actual as of 2024-02-12

**Pull request closed without merging**